lua/entities/gmod_wire_ledtape.lua

```diff
@@ -0,0 +1,341 @@
+		scale	= scaled texture height divided by original texture height ... (required with sprite)
+		backlit = draw fullbright base texture even when using a sprite? ..... (optional)
+		connect = connect the beams in the sprite texture? ................... (optional)
+	]]--
+
+	Wire_LEDTape.materialData = {
+		["fasteroid/ledtape01"] = {
+			name   = "5050 Sparse",
+			sprite = "fasteroid/ledtape01_sprite",
+			scale  = 512 / 904
+		},
+		["fasteroid/ledtape02"] = {
+			name   = "5050 Dense",
+			sprite = "fasteroid/ledtape02_sprite",
+			scale  = 512 / 434
+		},
+		["cable/white"] = {
+			name = "White Cable"
+		},
+		["arrowire/arrowire2"] = {
+			name = "Glowing Arrows",
+		},
+		["fasteroid/elwire"] = {
+			name   = "Electroluminescent Wire",
+			sprite = "fasteroid/elwire_sprite",
+			scale  = 256 / 2048,
+			backlit = true
+		},
+	}
+
+	local DEFAULT_SCALE = 0.5
+
+	local LIGHT_UPDATE_INTERVAL = CreateClientConVar( "wire_ledtape_lightinterval", "0.5", true, false, "How often environmental lighting on LED tape is calculated", 0 )
+
+	local LIGHT_DIRECTIONS = {
+		Vector(1,0,0),
+		Vector(-1,0,0),
+		Vector(0,1,0),
+		Vector(0,-1,0),
+		Vector(0,0,1),
+		Vector(0,0,-1)
+	}
+
+	local function getLitNodeColor(node)
+		if not node.lighting or node.nextlight < CurTime() then
+			local lightsum = Vector()
+			local pos = node[1]:LocalToWorld( node[2] )
+			for _, dir in ipairs(LIGHT_DIRECTIONS) do
+				lightsum:Add( render.ComputeLighting(pos, dir) )
+			end
+			lightsum:Mul( 1 / #LIGHT_DIRECTIONS )
+			node.lighting = lightsum:ToColor()
+			node.nextlight = CurTime() + LIGHT_UPDATE_INTERVAL:GetFloat()
+		end
+		return node.lighting
+	end
+
+
+	-- This system prevents calling LocalToWorld hundreds of times every frame, as it strains the garbage collector.
+	-- This is a necessary evil to prevent stuttering.
+	local LocalToWorld_NoGarbage_Ents = {}
+
+	local function LocalToWorld_NoGarbage(ent, pos)
+		ent.LEDTapeVecs = ent.LEDTapeVecs or {}
+		local LEDTapeVecs = ent.LEDTapeVecs
+
+		if ent.LEDTapeLastPos == ent:GetPos() and ent.LEDTapeLastAng == ent:GetAngles() and LEDTapeVecs[pos] then
+			return LEDTapeVecs[pos]
+		end
+
+		LEDTapeVecs[pos] = ent:LocalToWorld(pos)
+		LocalToWorld_NoGarbage_Ents[ent] = true -- update positions at the end
+
+		return LEDTapeVecs[pos]
+	end
+
+	local function LocalToWorld_NoGarbage_End()
+		for ent, _ in pairs(LocalToWorld_NoGarbage_Ents) do
+			if not IsValid(ent) then continue end
+			ent.LEDTapeLastPos = ent:GetPos()
+			ent.LEDTapeLastAng = ent:GetAngles()
+		end
+		LocalToWorld_NoGarbage_Ents = {}
+	end
+
+	hook.Add("PostDrawOpaqueRenderables","LEDTapeCleanup",LocalToWorld_NoGarbage_End)
+
+	local function DrawBeams(width, scrollmul, mater, path, getColor, extravertex)
+
+		if not IsValid(path[1][1]) then return end
+
+		local scroll = 0
+		local beam   = render.AddBeam
+
+		local beam2     = extravertex and beam or function() end -- branchless programming ftw
```

Comment:
Assuming lua doesn't optimize away creating this empty lambda, branchless isn't gonna help you here. I don't think It'd even help you for lua at all since it's so high level